bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -112,10 +168,85 @@ impl Mul<Weight> for FeeRate {
 impl Div<Weight> for Amount {
     type Output = FeeRate;
 
-    fn div(self, rhs: Weight) -> Self::Output { FeeRate(self.to_sat() * 1000 / rhs.to_wu()) }
+    fn div(self, rhs: Weight) -> Self::Output {
+        FeeRate { per_kwu: self * 1000 / rhs.to_wu() }
+    }
 }
 
-crate::parse::impl_parse_str_from_int_infallible!(FeeRate, u64, from_sat_per_kwu);
+#[cfg(feature = "serde")]
+pub mod serde {
+    #![allow(missing_docs)]
+    //! A module for serde-serializing fee rates in various units.
+
+    use serde::de::{Deserialize, Deserializer, Error};
+    use serde::ser::Serializer;
+
+    use super::{Amount, FeeRate};
+    use crate::amount::Denomination;
+
+    pub mod btc_per_kvb {
+        //! Serialize fee rates as bitcoin per kilo-virtual bytes.
+        use super::*;
+
+        pub fn serialize<S: Serializer>(r: &FeeRate, s: S) -> Result<S::Ok, S::Error> {
+            s.serialize_f64((r.to_per_vb_ceil() * 1000).to_btc())
+        }
+
+        pub fn deserialize<'de, D: Deserializer<'de>>(d: D) -> Result<FeeRate, D::Error> {
+            let btc_per_kvb = f64::deserialize(d)?;
+            let per_kwu = Amount::from_btc(btc_per_kvb / 4.0).map_err(D::Error::custom)?;
```

Comment:
Also please comment on the explicit decision to do the math here on the f64 instead of on the u64. Somehow my brain thinks that'd be more accurate, but I'm not 100% sure about that.

Reply:
In theory it can avoid some overflows, so you're right.